Gordon360/Services/RecIM/TeamService.cs

```diff
@@ -0,0 +1,176 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Models.ViewModels.RecIM;
+using Gordon360.Models.CCT.Context;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Gordon360.Authorization;
+
+namespace Gordon360.Services.RecIM
+{
+    public class TeamService : ITeamService
+    {
+        private readonly CCTContext _context;
+        private readonly IMatchService _matchService;
+
+        public TeamService(CCTContext context, IMatchService matchService)
+        {
+            _context = context;
+            _matchService = matchService;
+        }
+
+        public TeamViewModel GetTeamByID(int teamID)
+        {
+
+            var team = _context.Team
+                .Where(t => t.ID == teamID)
+                .Select(t => new TeamViewModel
+                {
+                    ID = teamID,
+                    Name = t.Name,
+                    Status = _context.TeamStatus
+                            .FirstOrDefault(s => s.ID == t.StatusID)
+                            .Description,
+                    Private = t.Private,
+                    Logo = t.Logo,
+                    Match = t.MatchTeam
+                        .Select(mt => _matchService.GetMatchByID(mt.MatchID)),
+                    Participant = t.ParticipantTeam
+                        .Select(pt => new ParticipantViewModel
+                        {
+                            Username = _context.ACCOUNT
```

Comment:
Make sure to pay attention to the GitHub warnings like the one below. This is warning you that _context.ACCOUNT could be null, which would cause errors at runtime.
```suggestion
                            Username = _context.ACCOUNT?
```

Reply:
This is probably the same misunderstanding on my end as below. Please check the FirstOrDefault operation on line 46.